addons/website/static/src/snippets/s_dynamic_snippet/000.js

```diff
@@ -149,12 +151,17 @@ const DynamicSnippet = publicWidget.Widget.extend({
      * @private
      */
     _getQWebRenderOptions: function () {
+        const dataset = this.$target[0].dataset;
+        const numberOfRecords = parseInt(dataset.numberOfRecords);
+        let numberOfElements;
+        if (config.device.isMobile) {
+            numberOfElements = parseInt(dataset.numberOfElementsSmallDevices) || DEFAULT_NUMBER_OF_ELEMENTS_SM
+        } else {
+            numberOfElements = parseInt(dataset.numberOfElements) || DEFAULT_NUMBER_OF_ELEMENTS;
+        }
```

Comment:
Could be a `const` by using a ternary operator. It is up to you, just a matter of taste.